### PR TITLE
fix: update type, remove unused imports, define AuthEvent type

### PR DESCRIPTION
**Pull request by cardenasj97**
## What kind of change does this PR introduce?

fix type, remove unused imports, fix comments

## What is the current behavior?

-

## What is the new behavior?

-

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: src/createServerClient.ts [hunk 1]
```diff
@@ -180,7 +173,7 @@ export function createServerClient<
     },
   );
 
-  client.auth.onAuthStateChange(async (event) => {
+  client.auth.onAuthStateChange(async (event: AuthEvent) => {
```
Comment (hf) [inline src/createServerClient.ts:176]: Hmm this type can be implicitly inferred.

Comment (cardenasj97) [reply src/createServerClient.ts]: Yes it is correct, but in this case TypeScript forces us to assign a type. I just assigned it appropriately by taking it from the supabase-js library

**Comment by cardenasj97** [comment]
This PR was already answered, can you take a look again @J0 @hf ?

**Review by J0 — APPROVED** [review]
Thanks @cardenasj97

**Pull request merged**